skymap_scanner/utils/data_handling.py

```diff
@@ -53,10 +54,24 @@ def stage_file(self, basename: str):
         local_destination_path = self.staging_path / basename
         http_source_path = f"{self.remote_path}/{basename}"
         # not sure why we use the -O pattern here
-        cmd = f"wget -nv -t 5 -O {local_destination_path} {http_source_path}"
-        return_value = os.system(cmd)
-        if return_value != 0 or not local_destination_path.is_file():
-            raise RuntimeError(f"Failed to retrieve data from remote source:\n-> {cmd}")
+        cmd = [
+            "wget",
+            "-nv",
+            "-t",
+            "5",
+            "-O",
+            str(local_destination_path),
+            http_source_path,
+        ]
+        try:
+            subprocess.run(cmd, check=True)
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError(f"Subprocess `wget` exited with status:\n -> {e}")
```

Comment:
You can use the `from e` syntax in addition or instead of using the f-string. I'm not sure if the `CalledProcessError` instance already provides the command. You may be able to get away with not excepting at all